infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -131,10 +187,21 @@ def restore_database(backup_fpath):
     )
     run_shell_command(create_db_cmd, env_vars=pg_env_vars)
 
+    # find the most recent backup or verify the specify backup exists
+    print("Getting backup file from S3")
+    session = create_s3_session()
+    if backup_obj_name is None:
+        backup_obj_name = get_most_recent_backup_obj_name(session)
+    elif backup_obj_name not in list_bucket_objects(session, BACKUP_BUCKET):
```

Comment:
Rather than rely on the global constant here, can we pass the bucket name as a parameter? You could make the default value of the parameter name equal to the constant. So the function signature would change to:

`def restore_database(..., backup_s3_bucket_name: Optional[str] = BACKUP_BUCKET). It's best to limit the number of global constants and variables that are used in functions so that it's as obvious as possible what the inputs/outputs of the function are when you call it.

I'm not super happy that I had this function just pull environment variables. At a minimum, I should have documented each of the env vars in the docstring. You don't have to do that, though. You're not the one who added that technical debt :D

Reply:
do you mean pass the backup_bucket name in as like a sys.argv? and should we just get rid of the global var altogether. Not sure what the best solution here looks like